contracts/token/ERC677Token.sol

```diff
@@ -1,77 +1,77 @@
-pragma solidity 0.5.17;
+pragma solidity ^0.8.9;
+pragma abicoder v1;
 
-import "@openzeppelin/contract-upgradeable/contracts/token/ERC20/ERC20Mintable.sol";
-import "@openzeppelin/contract-upgradeable/contracts/token/ERC20/ERC20Burnable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/presets/ERC20PresetMinterPauserUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/AddressUpgradeable.sol";
 import "./ERC677TransferReceiver.sol";
 import "./IERC677.sol";
 import "../core/Versionable.sol";
 
 /**
  * @dev reference from https://github.com/smartcontractkit/LinkToken
+
+
+  This contract is only used in tests, in the real protocol bridged tokens from the tokenbridge contracts are used
  */
-contract ERC677Token is IERC677, ERC20Burnable, ERC20Mintable {
-  string private _name;
-  string private _symbol;
-  uint8 private _decimals;
+contract ERC677Token is IERC677, ERC20PresetMinterPauserUpgradeable {
+  uint8 private configuredDecimals;
 
   function initialize(
-    string memory name,
-    string memory symbol,
-    uint8 decimals,
-    address minter
+    string memory _name,
+    string memory _symbol,
+    uint8 _decimals,
+    address owner
   ) public initializer {
-    _name = name;
-    _symbol = symbol;
-    _decimals = decimals;
-    initialize(minter);
+    // lots of initializers to override here, due to owner by default being msg sender
```

Comment:
There is no need really for this particular contract to be upgradeable--it's only ever used in our tests. If it makes your life easier, then let's just use a non-upgradeable ERC677